model/cluster_request.go

```diff
@@ -98,9 +102,27 @@ func (request *CreateClusterRequest) Validate() error {
 	}
 	// TODO: check zones and instance types?
 
+	if !contains(GetSupportedCniList(), request.Networking) {
+		return errors.Errorf("unsupported cluster networking option %s", request.Networking)
+	}
 	return nil
 }
 
+// GetSupportedCniList starting with three supported CNI networking options, we can add more as required
+func GetSupportedCniList() []string {
+	return []string{"amazon-vpc-routed-eni", "amazonvpc", "weave", "canal"}
```

Comment:
@Mshahidtaj I thought we were going to support `calico` too.

Reply:
Yeap agree. Can you specify the difference between amazon-vpc-routed-eni and amazonvpc?

Reply:
The amazon-vpc-routed-eni & amazonvpc both are same, The kops create cluster takes `--networking = amazon-vpc-routed-eni ` and kops get cluster shows as follows `networking:
    amazonvpc: {}`